### PR TITLE
#3266 Move contracts service and token price to shared

### LINKED ISSUE #4232: Update MyWalletWidget unit tests to not use sagas
In https://github.com/Neufund/platform-frontend/pull/4147 ` packages/web/app/components/dashboard/my-wallet/MyWalletWidget.spec.tsx` tests were skipped because they are starting sagas in `createIntegrationTestsSetup`. Since unit tests are only supposed to test a single piece of code, this suite and any other must be updated.

### PR DESCRIPTION
**Pull request by desfero**
**Tech notes**
It's not possible to fully share `contractsService` given `web3` and `ethers` generated typings for contracts are not fully interchangeable. I found a way to use an adapter in the cases where we want to adopt some contract to be used in `shared-modules` (check how `contractsService.rateOracle` is used in `token-price` module)

Also closes #4232 

Also, component tests dependent on the store where refactored. It was a huge pain to move modules to shared because you always end up dirty fixing unit tests in components

### CODE REVIEW

File: packages/web/app/components/dashboard/my-wallet/MyWalletWidget.spec.tsx [hunk 1]
```diff
@@ -1,61 +1,40 @@
...
-            neumarksDue: "2.01496624735285327461934e+23",
-            unlockDate: "1627528891",
-          },
-          neumarkAddress: "0x027a7a3991c4dd1dcb9db3f9a4dda8bab4d58f2f",
-          etherTokenBalance: "0",
-          euroTokenBalance: "0",
-          etherBalance: "5.8879038511742e+22",
-          neuBalance: "4.30223410549950281244511e+23",
-        },
+    const props = {
```
Comment (desfero) [inline packages/web/app/components/dashboard/my-wallet/MyWalletWidget.spec.tsx:24]: refactored to not use store but rather recieve props

File: packages/web/app/modules/init/sagas.ts [hunk 2]
```diff
@@ -1,26 +1,33 @@
...
+}
+
 function* initSmartcontracts({ web3Manager, logger }: TGlobalDependencies): any {
   try {
     yield fork(neuCall, initWeb3ManagerEvents);
     yield web3Manager.initialize();
 
     yield neuCall(initializeContracts);
-    yield neuCall(populatePlatformTermsConstants);
+    yield neuCall(watchTokenPrices);
```
Comment (desfero) [inline packages/web/app/modules/init/sagas.ts:30]: not sure if it should be here. Let me know what do you think

File: packages/wallet/app/store/create.tsx [hunk 3]
```diff
@@ -20,9 +22,13 @@ import { TAppGlobalState } from "./types";
 import { setupStorageModule } from "../modules/storage";
 
 export const createAppStore = (container: Container) => {
+  // TODO: Take universe address from artifacts meta.json
```
Comment (desfero) [inline packages/wallet/app/store/create.tsx:25]: Will do a ticket to find a way dynamically provide env variable (given we need to read file from disk and take the address from there)

File: packages/web/app/components/dashboard/my-portfolio/my-neu-widget/MyNeuWidget.spec.tsx [hunk 4]
```diff
@@ -1,250 +1,92 @@
-import { createMock, createMount, setupFakeClock, tid } from "@neufund/shared/tests";
-import { BigNumber } from "bignumber.js";
+import { createMount, tid } from "@neufund/shared/tests";
```
Comment (desfero) [inline packages/web/app/components/dashboard/my-portfolio/my-neu-widget/MyNeuWidget.spec.tsx:1]: Refactors the unit test to not be integration as it's impossible to maintain. Selectors are mocked so it's possible to change store shape without breaking unit tests for component

File: packages/web/app/components/dashboard/my-portfolio/my-neu-widget/MyNeuWidget.spec.tsx [hunk 5]
```diff
@@ -1,250 +1,106 @@
...
+  let stubs: Record<string, SinonStub>;
+
+  beforeEach(() => {
+    // stub button link as it uses store under the hood
+    stub(buttonLink, "ButtonLink").callsFake((props: any) => (
+      <div data-test-id={props["data-test-id"]} />
+    ));
+
+    stubs = {
+      selectIsLoading: stub(walletSelectors, "selectIsLoading").returns(undefined),
```
Comment (desfero) [inline packages/web/app/components/dashboard/my-portfolio/my-neu-widget/MyNeuWidget.spec.tsx:25]: let me know if you have an idea of how we can simplify mocking selectors?

File: packages/sagas/src/sagaEffects.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@ export * from "redux-saga";
 export { StringableActionCreator } from "@redux-saga/types";
 
 export * from "redux-saga/effects";
-export { call, race, SagaGenerator, select, getContext, take } from "typed-redux-saga";
+export { call, race, SagaGenerator, select, getContext, take, all } from "typed-redux-saga";
```
Comment (arjunu) [inline packages/sagas/src/sagaEffects.ts:6]: why not do `export *`?

File: packages/wallet/scripts/generate-contracts.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  try {
+    await generateKnownInterfaces();
+  } catch (e) {
+    console.error("Failed to read meta.json and generate knownInterfaces.json");
+    console.error(e.message);
+    process.exit(1);
+  }
+}
+
+main().catch(console.error);
```
Comment (arjunu) [inline packages/wallet/scripts/generate-contracts.js:56]: what's this catch for? there's already catch inside main

File: packages/web/test/integrationTestUtils.unsafe.tsx [hunk 8]
```diff
@@ -64,6 +65,11 @@ interface ICreateIntegrationTestsSetupOutput {
   sagaMiddleware: SagaMiddleware<{ container: Container; deps: TGlobalDependencies }>;
 }
 
+/**
+ * @deprecated Please don't write unit tests for components as integration tests.
```
Comment (arjunu) [inline packages/web/test/integrationTestUtils.unsafe.tsx:69]: 👍

File: packages/web/app/components/dashboard/my-wallet/MyWalletWidget.stories.tsx [hunk 9]
```diff
@@ -38,9 +38,9 @@ storiesOf("NDS|Molecules/Dashboard/MyWalletWidget", module)
     );
   })
   .add("default", () => (
-    <MyWalletWidgetComponentContainer {...props}>
+    <>
```
Comment (arjunu) [inline packages/web/app/components/dashboard/my-wallet/MyWalletWidget.stories.tsx:41]: is this required?

File: packages/sagas/src/sagaEffects.ts (hunk 6 above)
Comment (desfero) [reply packages/sagas/src/sagaEffects.ts:6]: to export * we need to refactor our app as the exports from `typed-redux-saga` (especially takeEvery and friends) do not yet in the same way as native `takeEvery`

File: packages/wallet/scripts/generate-contracts.js (hunk 7 above)
Comment (desfero) [reply packages/wallet/scripts/generate-contracts.js:56]: good point. leftover

File: packages/web/app/components/dashboard/my-wallet/MyWalletWidget.stories.tsx (hunk 9 above)
Comment (desfero) [reply packages/web/app/components/dashboard/my-wallet/MyWalletWidget.stories.tsx:41]: fixed, also on line `66`

**Pull request merged**